src/cloudai/configurator/base_agent.py

```
@@ -111,3 +112,38 @@ def update_policy(self, _feedback: Dict[str, Any]) -> None:
+
+        Returns:
+            int: Process-style return code (``0`` success, non-zero failure).
+            ``handle_dse_job`` accumulates this via ``err |= agent.run()``.
+        """
+        observation, _ = self.env.reset()
+        for _ in range(self.max_steps):
+            result = self.select_action(observation=observation)
+            if result is None:
+                break
+            step, action = result
+            logging.info(f"Running step {step} (of {self.max_steps}) with action {action}")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)